hazelcast/src/test/java/com/hazelcast/cluster/ClusterShutdownTest.java

```diff
@@ -56,11 +57,19 @@ private void testClusterShutdownWithSingleMember(final ClusterState clusterState
         final Thread thread = new Thread(new Runnable() {
             @Override
             public void run() {
-                instances[0].getCluster().shutdown();
+                while (true) {
+                    try {
+                        instances[0].getCluster().shutdown();
+                        break;
+                    } catch (Exception ignored) {
```

Comment:
there is an old saying, you know :)

"timeo hominem unius libri"

Joking aside, I fully understand your concern. As I stated, if cluster shutdown fails, it means that:
- Either cluster state failed to move into a shut-down-able (PASSIVE) state which creates no problem since everything will work fine. 
- Or cluster state is moved to (PASSIVE) state, some of the nodes has actually shut down but some others hasn't shutdown because of any failure (message loss etc.). In this case, still-running nodes must be shut-down by triggering the shutdown process on these nodes. Otherwise, they will not accept any operations since they are in PASSIVE state. This is still a safe state.